### DISCUSSION TITLE
Prevent MAIN.PY Running if BOOT.PY Aborts

### DISCUSSION THREAD

**Discussion by DS256** [micropython/micropython#10825] [RP2040 / Pico]
I've tried using sys.exit(1) from BOOT.PY but MAIN.PY still runs. I've looked at the 'machine' library but didn't find any 'stop' type function. Any suggestions or will MAIN.PY always run.

Thanks

**Comment by robert-hh** [comment]
You could try to raise an exception in boot.py.

**Reply by th3w** [reply]
but doesn't sys.exit() already do that?
[`
STATIC` mp_obj_t mp_sys_exit(size_t n_args, const mp_obj_t *args) {
    if (n_args == 0) {
        mp_raise_type(&mp_type_SystemExit);
](https://github.com/micropython/micropython/blob/8d9a7fd22836de6aa16c9d208212092805fc5641/py/modsys.c#L124)

**Reply by DS256** [reply]
Isn't that a 'C' example? I'm using micropython

**Reply by robert-hh** [reply]
Looking through the code, it seems that main.py is only skipped of there was force to soft reset in boot.py,like having Ctrl-D in a input stream. Error in the code do not stop it. Maybe the initial PR which introduced it was wrong.

**Reply by th3w** [reply]
It's part of the micropython core, where the method sys.exit() is handled (within  /py/modsys.c)

**Comment by DS256** [comment]
Ok, so it sounds like there is nothing in micropython/RPI Pico to handle this.

My fall back is for BOOT and MAIN to "create/check for" a CRASH.TXT file as a signal that an unrecoverable problem has occurred and requires manual intervention.

**Reply by robert-hh** [reply]
boot.py and main.py share the same global namespace. So it should be possible to just set a variable in boot.py which is checked in main.py

**Reply by DS256** [reply]
Brilliant. That works. Thanks
